ompi/proc/proc.h

```diff
@@ -71,10 +72,19 @@ BEGIN_C_DECLS
 struct ompi_proc_t {
     opal_proc_t                     super;
 
+#if OPAL_ENABLE_FT_MPI
```

Comment:
Structure size/order dependent on compilation options. Remove conditional compilation?
This moves the proc_endpoints array. This could break ABI compat?